comp/core/autodiscovery/providers/container_test.go

```diff
@@ -25,7 +26,7 @@ import (
 func TestProcessEvents(t *testing.T) {
 	store := fxutil.Test[workloadmetamock.Mock](t, fx.Options(
 		config.MockModule(),
-		logimpl.MockModule(),
+		fx.Provide(func() log.Component { return logmock.New(t) }),
```

Comment:
the `func() log.Component { return logmock.New(t) }` seems to be repeteded in a lot of place.
can we have a function in the `logmock` package that have directly the proper signature?